### PR TITLE
Remove/rename all bitwarden references

### PR DESCRIPTION
**Pull request by droslean**
/cc @openshift/openshift-team-developer-productivity-test-platform 
Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

**Comment by droslean** [comment]
`secret-bootstrapper-validation` job should fail.

**Comment by droslean** [comment]
/hold

A follow-up is required to change the `bootstrap` and `generator` config in the release repo.

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (alvaroaleman) [inline pkg/api/secretbootstrap/secretboostrap.go:22]: Can we maybe just do a first pr that adds new fields with new names, then change the config, then remove the old fields?

Comment (droslean) [reply pkg/api/secretbootstrap/secretboostrap.go:22]: I was hoping that I can do a follow-up, after the changes in this PR merge, in the release repo changing the config. It won't cost anything. Worst case scenario we will have a failed periodic job.

Comment (droslean) [reply pkg/api/secretbootstrap/secretboostrap.go:22]: I can disable the periodic jobs, land the changes in this PR and in release repo, then enable the jobs again. WDYT?

File: cmd/ci-secret-bootstrap/main.go
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main.go:82]: Maybe remove the mentioning of `bitwarden` here and below as well?

File: cmd/ci-secret-bootstrap/main.go
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main.go:281]: Why is the registry url resolving logic being removed?

File: cmd/ci-secret-bootstrap/main.go
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main.go:747]:
```suggestion
						errs = append(errs, fmt.Errorf("field %s in item %s doesn't exist", data.AuthField, data.Item))
```

File: cmd/ci-secret-bootstrap/main_test.go
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main_test.go:794]: Why was this test removed?

File: cmd/ci-secret-bootstrap/main_test.go
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main_test.go:864]: Why was this test removed?

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (alvaroaleman) [reply pkg/api/secretbootstrap/secretboostrap.go:22]: Not a fan of that, we also have presubmits that use this config and this is a gigantic PR that has a lot of potential to contain bugs, so I do not want to require it to be merged/reverted in lockstep with a config change. Please just leave the `json` annotation as it was, we can change it together with the config in a follow-up after this had some time to sink and we are confident it works correctly.

Also before this can merge, all usage of attachments and passwords in our config needs to be removed.

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (alvaroaleman) [inline pkg/api/secretbootstrap/secretboostrap.go:34]: Same comment here regarding not changing the json annotations.

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (alvaroaleman) [inline pkg/api/secretbootstrap/secretboostrap.go:35]: Why was this removed, do we not use it?

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (droslean) [reply pkg/api/secretbootstrap/secretboostrap.go:22]: Yea you are right. It's safer to do it like this. I will update the PR.

File: cmd/ci-secret-bootstrap/main.go
Comment (droslean) [reply cmd/ci-secret-bootstrap/main.go:281]: First we had only the registry-url field that it was supposed to be a field in bitwarden, but we added the `registryURL` that gets the value directly from the config. We kept the old one to keep backwards compatibility and now it is not being used at all. So it is safe to remove it as well.

File: cmd/ci-secret-bootstrap/main_test.go
Comment (droslean) [reply cmd/ci-secret-bootstrap/main_test.go:864]: `RegistryURLBitwardenField` is removed, so there is no need for that validation at all.

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (alvaroaleman) [inline pkg/api/secretbootstrap/secretboostrap.go:22]: Why did we add a new field? Please just rename the existing field without changing the json annotation

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (alvaroaleman) [inline pkg/api/secretbootstrap/secretboostrap.go:34]: Please leave the json tag as it was

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (droslean) [reply pkg/api/secretbootstrap/secretboostrap.go:22]: I wanted to create a new json but still support the old one to keep backwards compatibility, and change the config without breaking anything.

File: pkg/api/secretgenerator/testdata/zz_fixture_TestLoadConfigFromPath_single_parameter_with_multiple_values.yaml
Comment (alvaroaleman) [inline pkg/api/secretgenerator/testdata/zz_fixture_TestLoadConfigFromPath_single_parameter_with_multiple_values.yaml:1]: I think I would feel more comfortable if we didn't change any existing fixtures to proof that the current as we have it right now will continue to work (only add some more to show that the new config will also work)

File: pkg/api/secretbootstrap/secretboostrap.go
Comment (alvaroaleman) [reply pkg/api/secretbootstrap/secretboostrap.go:22]: Ah sorry, I missed that, the fact that you changed all existing fixtures and tests to use the new format confused me. Can we not do that to:
* Reduce the size of this PR
* Proof that the old format our config is in still works?

Comment (droslean) [reply pkg/api/secretbootstrap/secretboostrap.go:22]: Well, eventually we will end up changing them, so in this PR I only change their names and load them from the old values during the unmarshaling. The old version should be working fine.  I can try to see what it takes to make this PR smaller and better reviweablt.

File: cmd/ci-secret-bootstrap/main_test.go
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main_test.go:109]: Can we not change anything here to show that the existing config works as before?

Comment (droslean) [reply cmd/ci-secret-bootstrap/main_test.go:109]: Let me add both cases then.

File: cmd/ci-secret-bootstrap/main_test.go
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main_test.go:109]: Why is this conversation resolved? All the tests got changed

Comment (droslean) [reply cmd/ci-secret-bootstrap/main_test.go:109]: I added below an item using the old fields as well. That should be enough, right?

**Comment by droslean** [comment]
@alvaroaleman Is there anything left here? Sorry I didn't focus on this PR lately.

File: cmd/ci-secret-bootstrap/main_test.go
Comment (alvaroaleman) [reply cmd/ci-secret-bootstrap/main_test.go:109]: Please just leave the existing tests unmodified and add new ones for the new functionality.

Comment (droslean) [reply cmd/ci-secret-bootstrap/main_test.go:109]: Got it. I'll make the changes. Thanks.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**
